### ISSUE TITLE
Manage static asset with bower and grunt

### ISSUE THREAD

**Issue by gianarb** [gianarb/zfbp#1]
[Bower](http://bower.io) is a famous package manage for web project, it's used to manage frontend dependencies.

[GruntJS](http://gruntjs.com) is a Javascript task manage it support minified and lint of css, javascript, less compiled.. The idea is to use it to manage build of fronted ZF application

**Comment by gianarb** [comment]
How can I config files path in only one part of application?
I use this list to build two different layout
* dev layout with a list of all js and css file not min
* prod file with compress.css and compress.min.js

**Comment by stefanotorresi** [comment]
@gianarb you can use `grunt.option()` to define an arbitrary CLI option, and then use it in your tasks configuration, so `grunt build --prod` will run the same tasks but (for example) with different file sets.

**Comment by gianarb** [comment]
@stefanotorresi the different files for different env works.. But I don't know how write the list of files so that use it for grunt and for append it into the `layout.phtml` files
```json
{
    "js_vendor": [
        "public/bower_components/jquery/dist/jquery.js",
        "public/bower_components/jquery/dist/js/bootstrap.js"
    ],
    "css_vendor": [
        "public/bower_components/jquery/dist/css/bootstrap.css"
    ],
}
```
I can read it by Grunt to manage min ecc.. And I can use it into layout.phtml to append correct file in dev mode..

In production in this case I append this file [JS](https://github.com/gianarb/zfbp/pull/2/files#diff-35b4a816e0441e6a375cd925af50752cR11) and [CSS](https://github.com/gianarb/zfbp/pull/2/files#diff-35b4a816e0441e6a375cd925af50752cR21)

My point is, ZF conf in php.. but list of asset in json... :facepunch:

**Comment by stefanotorresi** [comment]
I see, well I usually have just one js file and one css file and keep it always the same in `layout.phtml`, and let grunt handle the content (i.e. minified or not).

**Comment by gianarb** [comment]
mmm for example [append by view helper](https://github.com/zendframework/ZendSkeletonApplication/blob/master/module/Application/view/layout/layout.phtml#L15) 

```php
<?php 
    if($this->env == "prod"){ 
        $this->headLink()->prependStylesheet($this->basePath() . '/bin/output.min.css')
    } else {
        // see NOTE
    }
?>
```
NOTE: I cycle  [this array](https://github.com/gianarb/zfbp/pull/2/files#diff-35b4a816e0441e6a375cd925af50752cR22)  and append it..
But for read this list of file in grunt and in PHP, I can write it into JSON file.. I don't see any alternative :bomb:

**Comment by stefanotorresi** [comment]
yep, got it, i just don't like it. i'd rather link always the same files in PHP, and then let Grunt handle whether to minify the assets or not.

to be clear:

```php
// php
$this->headLink()->prependStylesheet($this->basePath('bin/styles.css'))
$this->headScript()->prependFile($this->basePath('bin/scripts.js'));
```

```js
// grunt
var prod = grunt.options('prod');
grunt.registerTask('bin', prod ? ['uglify', 'cssmin'] : ['concat']);
```

or something like that.

**Comment by gianarb** [comment]
aaaaaa ok! Now I got it! :+1:

**Comment by samsonasik** [comment]
i'm +1 for use same target for whatever the environment selected. on using tools, i think it need more readme on how to start so more user can use its feature.
